packages/elements/src/time-picker/index.ts

```diff
@@ -727,23 +734,17 @@ export class TimePicker extends ControlElement {
    * @returns {void}
    */
   private changeValueBy(amount: number, segment: Segment): void {
-    let offset = 0;
-    switch (segment) {
-      case Segment.HOURS:
-        offset = this.hours === null ? 0 : amount * MILLISECONDS_IN_HOUR;
-        break;
-      case Segment.MINUTES:
-        offset = this.minutes === null ? 0 : amount * MILLISECONDS_IN_MINUTE;
-        break;
-      case Segment.SECONDS:
-        offset = this.seconds === null ? 0 : amount * MILLISECONDS_IN_SECOND;
-        break;
-      // no default
+    const segmentTarget = this[segment];
+    const segmentMap = SegmentMap[segment];
```

Comment:
I prefer to use the destructuring assignment because the source code will be concise.

```js
`const { miliseconds, cycle } = SegmentMap[segment];
```